x-pack/plugins/fleet/public/applications/integrations/sections/epm/screens/detail/policies/package_policies.tsx

```diff
@@ -109,12 +114,12 @@ export const PackagePoliciesPage = ({ name, version }: PackagePoliciesPanelProps
       const updatableIntegrationRecord = updatableIntegrations.get(
         packagePolicy.package?.name ?? ''
       );
-
       const hasUpgrade =
         !!updatableIntegrationRecord &&
         updatableIntegrationRecord.policiesToUpgrade.some(
-          ({ id }) => id === packagePolicy.policy_id
+          ({ pkgPolicyId }) => pkgPolicyId === packagePolicy.id
```

Comment:
This is line is the fix to disable the button

Reply:
So policy_id was an incorrect field?

Reply:
Yes, updatableIntegration.policiesToUpgrade contains information about the agent policy (Default policy in this case) and the package policy (which has an upgrade available), where `id` is the ID of the agent policy and `pkgPolicyId` is the ID of the package policy.

Previously we were incorrectly comparing the agent policy IDs so every package policy in an agent policy would show as needing to upgrade. I have changed  it so that we compare the package policy ID so that only the out of date packages show as needing an upgrade.